### ISSUE TITLE
Rule `property-require-support`

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#2004]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Require use `support` for property (properties). We love `flex`, but some browser not support their or have not fully support (example UC browser or old Android 4.1, 4.2 not support `flex-wrap` and `flex-shrink`). Sometimes we forget about this, be good have rule which reminds us of this. Also be good be able to specify more than one property such as an array (use `flex` is good, but if you use `flex` and `flex-wrap` need require `support`).

> Which rule, if any, is this issue related to?

`property-require-support`

> What CSS is needed to reproduce this issue?

no matter

> What stylelint configuration is needed to reproduce this issue?

no matter

> Which version of stylelint are you using?

no matter

> How are you running stylelint: CLI, PostCSS plugin, Node API?

no matter
]
> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no matter

> What did you expect to happen?

no matter

> What actually happened (e.g. what warnings or errors you are getting)?

no matter

**Comment by davidtheclark** [comment]
@evilebottnawi Sounds interesting. Can you provide some examples of what would trigger a warning, and what would not?

**Comment by evilebottnawi** [comment]
@davidtheclark 
`property-require-support: [{'display': 'flex'}]`
```css
a {
  display: flex; /* Error here */
}
```

`property-require-support: [{'display': 'flex', 'display': 'flex-wrap'}]`
```css
a {
  display: flex; /* No error */
}
```
```css
a {
  display: flex;
  flex-wrap: wrap; 
}
```
`property-require-support: [{'display': 'flex', 'display': 'flex-wrap'}, {'text-align-last': '*'}]`
```css
a {
  text-align-last: justify; /* Error here */
}
```
`property-require-support: [{'flex(-*)?': '*'}]`
All using `flex` property and `flex-*` properties for all values.

**Comment by davidtheclark** [comment]
@evilebottnawi Your examples are confusing me because none of them have `@supports`. Is that not what you're talking about?

**Comment by evilebottnawi** [comment]
@davidtheclark I gave an example of errors. That it is not clear?

**Comment by davidtheclark** [comment]
@evilebottnawi None of these examples show the use of `@supports`. Am I right in thinking that `@supports` is what this rule is all about?

**Comment by evilebottnawi** [comment]
@davidtheclark oh sorry
`property-require-support: [{'display': 'flex'}]`
Error:
```css
a {
  display: flex;
}
```
Not error:
```
a {
  @support (display: flex) {
    display: flex;
  }
}
`property-require-support: [{'display': 'flex', 'flex-wrap': 'wrap'}]`
Error:
```css
a {
  @support (display: flex) and (flex-wrap: wrap) {
    display: flex;
    flex-wrap: wrap;
  }
}
```
No error:
```css
a {
  display: flex; 
}
```

```css
a {
  flex-wrap: wrap;
}
```
Also support regex in property and value. Not sure of the groups of clarity is probably better to avoid them, but leave is here to discuss

**Comment by davidtheclark** [comment]
@evilebottnawi Is your second example reversed? Shouldn't the one with `@support` *not error*?

**Comment by evilebottnawi** [comment]
@davidtheclark yep, I'm mixed in second example, first not error, second and third with error

**Comment by davidtheclark** [comment]
@evilebottnawi If you wouldn't mind, I think this would be more clear if you closed the issue and opened a new one with accurate examples right at the top. Currently there's a lot of confusing stuff.

**Comment by evilebottnawi** [comment]
@davidtheclark ok, i am do this in the evening today, sorry little time :sob:
